pkg/serving/config_changes.go

```diff
@@ -91,31 +92,46 @@ func UpdateConcurrencyLimit(template *servingv1.RevisionTemplateSpec, limit int6
 	return nil
 }
 
-// UnsetUserImageAnnot removes the user image annotation
-func UnsetUserImageAnnot(template *servingv1.RevisionTemplateSpec) {
+// UnsetUserImageAnnotation removes the user image annotation
+func UnsetUserImageAnnotation(template *servingv1.RevisionTemplateSpec) {
 	delete(template.Annotations, UserImageAnnotationKey)
 }
 
-// SetUserImageAnnot sets the user image annotation if the image isn't by-digest already.
-func SetUserImageAnnot(template *servingv1.RevisionTemplateSpec) {
+// UpdateUserImageAnnotation sets the user image annotation if the image isn't by-digest already.
+func UpdateUserImageAnnotation(template *servingv1.RevisionTemplateSpec) {
 	// If the current image isn't by-digest, set the user-image annotation to it
 	// so we remember what it was.
-	currentContainer, _ := ContainerOfRevisionTemplate(template)
-	ui := currentContainer.Image
-	if strings.Contains(ui, "@") {
-		prev, ok := template.Annotations[UserImageAnnotationKey]
+	currentContainer, err := ContainerOfRevisionTemplate(template)
+	if err != nil {
+		return
```

Comment:
Shouldn't we propagate the actual error?

Reply:
Indeed, if an error is returned it should be propagated. However, I think returning an error from `ContainerOfRevisionTemplate` is fishy anyway, as a nil-container is a valid outcome anyway and the only case, when error is returned, is when the template does not has a container. I remover the error slot therefore and did some addition minor refactorings.